shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.h

```diff
@@ -8,22 +8,45 @@
 #import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterKeySecondaryResponder.h"
 
+@class FlutterTextField;
+
 /**
  * A plugin to handle text input.
  *
  * Responsible for bridging the native macOS text input system with the Flutter framework text
  * editing classes, via system channels.
  *
  * This is not an FlutterPlugin since it needs access to FlutterViewController internals, so needs
- * to be managed differently.
+ * to be managed differently. The FlutterViewController has the ownership of this plugin.
+ *
+ * When accessibility is on, accessibility bridge creates a native text field, i.e.
+ * FlutterTextField, for every text field in the Flutter. This plugin acts as a field editor for
```

Comment:
If we only use one nstextfield to represent every flutter textfields, we would have to change the rect of it when focus move to the next flutter textfield. We only know the flutter textfield is focused after the focus has shifted to the node, so there may be one frame delay until the rect will be updated. I will try to prototype it and see whether there is any visual glitch if we do that

Reply:
If you keep this approach, you should test performance in extreme cases, such as scrolling a list view full of rows where each row has several text fields, just to be sure it's not going to be a perf issue.

Reply:
@LongCatIsLooong - what are we doing these days for text input on iOS? I know for a long time we used one backing textfield and moved it around as needed. Eventually we went to two so that one could be for secure/password input. IIRC, we do the same for a11y text input too, @gaaclarke may know more there.

Reply:
For iOS the UITextInput doesn't need to be focus in a11y for it to work as long as it is the primary responder, so we just have one textInput(except secure) that represents the current focused flutter text field. We don't need to move it around because it won't get focus.

Reply:
I can't think of a way to use one single NSTextField. If two TextFields uses the same NSTextField as its NSAccessibilityElement, I can't think of a way to detect the focus shift from one to another because it is the same NSTextField that gets the focus. I do a little bit of stress test, and it seems to be fine if there are about ~100 textfields.